### ISSUE TITLE
Re: Comments / Whitespace - standalone delimited comments left.

### ISSUE THREAD

**Issue by swizca** [penibelst/jekyll-compress-html#91]
I appreciate:
> compress_html:
>  comments: ["\"]
> Whitespaces around the tags prevent conditional comments from being deleted.

So I have, for example, made some comments like this `<!--|  comment |-->`.

`<!--  comment -->`'s are indeed deleted, however ``, where bol/eol = beginning / end of line, are still present.

Shouldn't the whitespaced stripped comment delimiters (also) be used?

[Even if `<--{space}{eol}` is used, editors frequently strip whitespace from {eol}, defeating this intent.]

**Comment by jnvsor** [comment]
Good point. IIRC I had a way to fix that once but it tanked performance. For a quick fix you could try something like:

```
<!-- .
comment
. -->
```

**Comment by swizca** [comment]
> Or you could change it to <!---

Reverses the norm. \<--| is trying to except certain comments, not force one to re-jigger all comments except the ones one wants to keep. Let alone imported code ...

Almost feels like a reverse-comments (keep_comments) setting would be useful.

**Comment by swizca** [comment]
Actually ... the same occurs with \ has no impact).

So, it would seem that a provided comment: should be preprocessed  to munge any whitespace to mean all whitespace.

Or ... the provided comment: to accept / presume RE's. (?)

This would allow \<!--[^\|] , in these examples, and \<!--[^\\[] would accommodate the current documented edge case. (Conditional if's.)

No?

**Comment by jnvsor** [comment]
That preprocessing is what's going to kill your performance, and liquid doesn't support regex. (If it did this project would be way way easier :P)

A better solution would be to look for `<!--` and ensure it's *not* followed by `[if` but I dread to think what it'll do to perf

**Comment by swizca** [comment]
> liquid doesn't support regex.

Would seem to be the operative problem.

Unless one could introduce / intervene an additional interpreter. e.g. grep scripts. Good luck getting that into a github approved plugin. Interesting, here, with html_compress, is the level of flexibility offered - unlike jekyll-tidy (htmlbeautifier / htmlcompressor) - which have few / no options.

> A better solution would be to look for "] seems needed.

And, given that ' -->' or '-->' appear to have no impact, I wonder if specifying the end comment delimiter is actually useful. Always take (or not) from the given comment starter to '-->'? [One may want to consistently use a matching comment ender, but that is a writing convention, not a coding requirement?]

**Comment by jnvsor** [comment]
> Either a keep comment delimiter keep_comment_delims: ["[" "]"] or a comment exceptor keep_comments: [""] seems needed.

Well yes but how would you implement this? This is liquid! It's the most annoying "Language" to do anything complicated in!

Currently comments, pre tags and the like are implemented by string splitting since that's the only performant way to find a string in another one in liquid (The alternative being to loop character by character that more than triples runtime IIRC)
